dashboard/app/models/regional_partner.rb

```diff
@@ -33,6 +34,57 @@ class RegionalPartner < ActiveRecord::Base
 
   has_many :pd_workshops
 
+  # Make sure the phone number contains at least 10 digits.
+  # Allow any format and additional text, such as extensions.
+  PHONE_NUMBER_VALIDATION_REGEX = /(\d.*){10}/
+
+  ADDRESS_NOT_VERIFIED = "Address could not be verified. Please double-check."
+  DOES_NOT_MATCH_ADDRESS = "doesn't match the address. Did you mean"
+  INVALID_STREET_ADDRESS = "must be a valid street address (no PO boxes)"
+
+  validates :name, length: {minimum: 1, maximum: 255}
+  validates :group, numericality: {only_integer: true, greater_than: 0}, if: -> {group.present?}
+  validates_format_of :phone_number, with: PHONE_NUMBER_VALIDATION_REGEX, if: -> {phone_number.present?}
+  validates_inclusion_of :state, in: STATE_ABBR_WITH_DC_HASH.keys.map(&:to_s), if: -> {state.present?}
+  validate :valid_address?, if: -> {street.present? && address_fields_changed?}
+
+  def valid_address?
+    # only run this validation once others pass
+    return unless errors.empty?
+
+    found = Geocoder.search(full_address)
+    if found.empty?
+      errors.add(:base, ADDRESS_NOT_VERIFIED)
+    else
+      if found.first.postal_code != zip_code
+        errors.add(:zip_code, "#{DOES_NOT_MATCH_ADDRESS} #{found.first.postal_code}?")
+      end
+      if found.first.state_code != state
+        errors.add(:state, "#{DOES_NOT_MATCH_ADDRESS} #{found.first.state_code}?")
+      end
+      unless found.first.street_number
+        errors.add(:street, INVALID_STREET_ADDRESS)
+      end
+    end
+  end
```

Comment:
Are we sure we want to fully validate the address here? Occasionally real addresses aren't found correctly by the geocoder, and especially don't return with a street number. In the workshop material order [form](https://github.com/code-dot-org/code-dot-org/blob/e38b71109b55f218afa5c8548146354e6c56d0c1/dashboard/app/views/pd/workshop_material_orders/new.html.haml#L24) we added a [manual override](https://github.com/code-dot-org/code-dot-org/blob/735a3dce8f0d77dd02b5d63d6eed7386f9692449/dashboard/app/models/pd/workshop_material_order.rb#L66) "I verify that this address is correct, and I want to use it to ship my order" checkbox so teachers with "invalid" (according to the geocoder) street addresses could still order supplies. 

Do we care in this context if it's a PO box? Do we care to have such strict validation? If so we should at least provide an override.

Check with @tanyaparker but I don't think we need any full address validation here, beyond basic formatting.